zebra-state/src/sled_state.rs

```diff
@@ -78,11 +78,23 @@ impl FinalizedState {
         };
 
         if let Some(tip_height) = new_state.finalized_tip_height() {
-            new_state.stop_if_at_height_limit(
-                StopCheckContext::OnLoad,
-                tip_height,
-                new_state.finalized_tip_hash(),
-            );
+            let tip_hash = new_state.finalized_tip_hash();
+            if new_state.is_at_stop_height(tip_height, tip_hash) {
+                let debug_stop_at_height = self
+                    .debug_stop_at_height
+                    .expect("true from `is_at_stop_height` implies `debug_stop_at_height` is Some");
+
+                if tip_height > debug_stop_at_height {
+                    tracing::error!(
+                        ?debug_stop_at_height,
+                        ?tip_height,
+                        ?tip_hash,
+                        "previous state height is greater than the stop height",
+                    );
+                }
+
+                std::process::exit(0);
```

Comment:
```suggestion
                // There's no need to sync before exit, because the trees have just been opened
                std::process::exit(0);
```